localstack/services/cloudformation/engine/template_deployer.py

```diff
@@ -450,33 +448,44 @@ def _resolve_refs_recursively(stack_name, resources, value: dict | list | str |
             item_to_sub[1].update(attr_refs)
 
             for key, val in item_to_sub[1].items():
-                val = resolve_refs_recursively(stack_name, resources, val)
+                val = resolve_refs_recursively(stack_name, resources, mappings, val)
                 result = result.replace("${%s}" % key, val)
 
             # resolve placeholders
-            result = resolve_placeholders_in_string(result, stack_name, resources)
+            result = resolve_placeholders_in_string(result, stack_name, resources, mappings)
             return result
 
         if stripped_fn_lower == "findinmap":
-            attribute = value[keys_list[0]][1]
-            attr = resolve_refs_recursively(stack_name, resources, attribute)
-            selected_map = value[keys_list[0]][0]
-            if isinstance(selected_map, dict) and "Ref" in selected_map:
-                selected_map = resolve_ref(stack_name, resources, selected_map["Ref"], "Ref")
-            result = resolve_ref(stack_name, resources, selected_map, attribute=attr)
-            if not result:
+            # "Fn::FindInMap"
+            mapping_id = value[keys_list[0]][0]
+
+            if isinstance(mapping_id, dict) and "Ref" in mapping_id:
+                # TODO: ??
+                mapping_id = resolve_ref(stack_name, resources, mappings, mapping_id["Ref"], "Ref")
```

Comment:
a bit of context for the ?? here: I'm just not a fan of using resolve_ref here (because not all refs here are valid) in the long term but for now its fine.